src/app/ipv/middleware.js

```diff
@@ -310,10 +310,11 @@ module.exports = {
       if (pageRequiresUserDetails(pageId)) {
         const userDetailsResponse =
           await coreBackService.getProvenIdentityUserDetails(req);
-        renderOptions.userDetails = generateUserDetails(
-          userDetailsResponse,
-          req.i18n,
-        );
+
+        const userDetails = generateUserDetails(userDetailsResponse, req.i18n);
+
+        renderOptions.userDetails = userDetails;
+        req.session.userDetails = userDetails;
```

Comment:
How long do the frontend sessions last? Do you think there's any risk to us storing/using this (PII) data inappropriately?